plumbing/transport/fetch_pack.go

```diff
@@ -128,6 +128,28 @@ func (r *UploadPackRequest) Have(h ...plumbing.Hash) {
 	r.Haves = append(r.Haves, h...)
 }
 
+func (r *UploadPackRequest) IsEmpty() bool {
+	if len(r.Haves) < len(r.Wants) {
+		return false
+	}
+
+	for _, h := range r.Wants {
+		found := false
+		for _, oh := range r.Haves {
+			if h == oh {
+				found = true
+				break
+			}
+		}
```

Comment:
Ok. Adding the areSetSubset.

r.Wants and r.Haves need to be public (with the current code). Let's revisit this in the next PR (refactor of UploadPack messsages).